spec/models/infra_conversion_job/dispatcher_spec.rb

```diff
@@ -0,0 +1,31 @@
+RSpec.describe InfraConversionJob::Dispatcher do
+  let(:zone) { FactoryBot.create(:zone) }
+  let(:dispatcher) do
+    described_class.new.tap do |dispatcher|
+      dispatcher.instance_variable_set(:@zone_name, zone.name)
+    end
+  end
+
+  before do
+    @server = EvmSpecHelper.local_miq_server(:name => "test_server_main_server", :zone => zone)
+  end
+
+  describe '.waiting?' do
```

Comment:
Should we have a test cases for "no items in the queue" and "multiple items in the queue (perhaps with different states)"?

Reply:
Not a bad idea, these specs are just copied from the existing job_proxy_dispatcher spec but only testing for the specific job types now